src/services/tripsService.js

```diff
@@ -83,6 +84,41 @@ class tripHelper {
 
     return allTrips;
   }
+
+  /**
+   *
+   * @returns {object} most traveled centres infos
+   */
+  static async getTrendingCentres() {
+    const trendingCentres = [];
+    await travelRequests.findAll({
+      attributes: [[Sequelize.fn('DISTINCT', Sequelize.col('to')), 'centre']]
+    }).map(async (row) => {
+      const destination = row.dataValues.centre;
+      const { count, rows } = await travelRequests.findAndCountAll({
+        where: {
+          to: destination,
+        },
+        order: [
+          ['createdAt', 'DESC']
+        ]
+      });
+      const approvedRequests = rows.filter((trip) => trip.status === 'approved');
+      const pendingRequests = rows.filter((trip) => trip.status === 'pending');
+      const rejectedRequests = rows.filter((trip) => trip.status === 'rejected');
+      const accommodationBookedTrips = rows.filter((trip) => trip.accommodation === 'yes');
+      const singleResponse = {
+        allTripsRequested: count,
+        approvedTrips: approvedRequests.length,
+        pendingTrips: pendingRequests.length,
+        rejectedTrips: rejectedRequests.length,
+        tripsBookedAccommodation: accommodationBookedTrips.length
+      };
+      trendingCentres.push({ [destination]: singleResponse });
+    });
+
+    return trendingCentres;
+  }
```

Comment:
please note that it's not about travel requests it's about accommodations and to narrow it down it's all about the most travelled accommodation. to those details can be found in the bookings table 

read the story and let me know if I got it wrong
https://www.pivotaltracker.com/story/show/170947468

Reply:
@blaise82 Thank you for addressing this;

Honestly, before working on this feature, I've spent a couple of hours reading ```PT story ``` for the sake of getting a clue on how to go about this functionality. The story says:

```
Why
Users should be able to get info on the most frequently traveled centre/destination

As a user or manager, I should get information on the most traveled centre/destination so that logical conclusions can be drawn.

```
So I thought if we get like stats of how every destination was traveled to, that will make sense to someone who wants to make a decision of where to travel to. Hence that 's why I constructed a query to fetch trending destinations in ``travelRequests``` table and collected some useful infos like 
```
  all trip requests that were made,
   which trip requests were approved,
   Which ones are still pending,
   Which ones were rejected,
   Which ones requested accomondations
```
Above all, if there is another better way of doing this, please share with me those ideas.

Thanks in advance

Reply:
@blaise82 Thank you for that, Let go and figure out why and I'll get back you when I am done

Reply:
@blaise82 I'd like to let you know that I've finished implementing the other feedback like so